nanopub/publication.py

```diff
@@ -28,10 +29,11 @@ class Publication:
         source_uri (str): The URI of the nanopublication that this Publication represents (if
             applicable)
         introduces_concept: The concept that is introduced by this Publication.
-
+        signed_with_public_key: The public key that this Publication is signed with.
+        is_test_publication: Whether this is a test publication
     """
 
-    def __init__(self, rdf=None, source_uri=None):
+    def __init__(self, rdf, source_uri=None):
```

Comment:
Should we give a warning/error if the ```rdf``` passed is e.g. ```None```?

Reply:
This now gives an error if you don't specify it, which is already a huge step forward. 

Checking that the `rdf` is not `None` would be nice, we could also check that it is an instance of `rdflib.Graph`. We don't really do much type & value checking anywhere throughout the library, but we could. Maybe you should create an issue if you want to do this (but I'm fine with not doing it)

I've added type hinting so it is clear what to pass which I think is enough in this case. We don't expect the user to use this actually.